src/backend/api/meals.js

```diff
@@ -1,15 +1,63 @@
-import express from "express";
+import express, { response } from "express";
 const router = express.Router();
 import knex from "../database.js";
 
 router.get("/", async (request, response) => {
   try {
     // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
+    const titles = await knex("meal").select("title");
     response.json(titles);
   } catch (error) {
     throw error;
   }
 });
 
+// maxPrice	Number	Returns all meals that are cheaper than maxPrice.	api/meals?maxPrice=90
+router.get("/", async (request, response) => {
+  try {
+    const maxPrice = await knex("meal").where("price", "<", 90);
+    response.json(maxPrice);
+  } catch (error) {
+    throw error;
+  }
+})
+
+// availableReservations	Boolean	Returns all meals that still have available spots left, if true. If false, return meals that have no available spots left.1	api/meals?availableReservations=true
+router.get("/", async (request, response) => {
```

Comment:
These endpoints do not seem to be fully finished. 🥺  I think binding the same path, i.e. `GET /`, multiple times simply overrides and they do not take into account the query parameters. However, it seems you generally got the idea down for how to accomplish each type of filtering! ⭐ 

A big chunk of the challenge comes from supporting multiple of the filters as once. It can be quite a mouthful to get right, so feel free to reach out if you have issues. 😄

Reply:
I guess I should implement object destructuring 👀

Reply:
this works:
```router.get("/", async (req, res) => {
  let {
    maxPrice,
    availableReservations,
    title,
    dateAfter,
    dateBefore,
    limit,
    sortKey,
    sortDir
  } = req.query;

  maxPrice = await knex('meal').select('price').where('price', '<', req.query.maxPrice);
  res.json(maxPrice);

})
```